allennlp/models/crf_tagger.py

```diff
@@ -223,8 +228,13 @@ def forward(
             output["top_k_tags"] = best_paths
 
         if tags is not None:
+            if ignore_loss_on_o_tags:
+                o_tag_index = self.vocab.get_token_index("O", namespace=self.label_namespace)
```

Comment:
Are `"O"` tags a general-enough concept to hard-code it like this?

I'd rather have a flag named `ignore_loss_tags` that's a set of strings. Then you can use it to tag anything.

Reply:
Yes, I believe "O" is general enough for this.  Any flavor of BIO tagging with have this tag, and you really only want this kind of behavior when you have a span prediction model that uses BIO tags.

Reply:
You would never use this for POS tagging or some other tagging that's not span based?

Reply:
It's hard for me to imagine a situation where that's useful.  And even in a POS tagging scenario, a list of strings wouldn't solve the problem, because if there are two matching POS tags in the sentence, you couldn't exclude just one of them.  That approach needs a different solution, and it's a use case that I have a hard time imagining anyway, so it didn't seem like a good idea to design for it.

Reply:
The way I'm really coming at it from is that this component is called "CRF Tagger", but we're putting in code that only works when you use it as a "CRF Span Tagger" (using magic strings that you just have to know about, no less). The examples are just me trying to boil down my unease to something specific.

It seems pretty arbitrary to give the string `"O"` different handling from all other strings, doesn't it?

I guess I can agree to it by thinking of it as something that may help you in a common case, and you can otherwise ignore it 🤷. I just worry that it becomes a maintenance problem when we discover people actually do want to tag things that aren't spans.

Reply:
https://github.com/allenai/allennlp/blob/master/allennlp/models/crf_tagger.py#L39-L51

All of this code is specific to span tagging.  I'm just adding one more flag that's specific to span tagging, which you can ignore if you're not doing span tagging.  And all of the supported label encodings assign specific meaning to "O", so this isn't exactly arbitrary.  The flag itself also gives the string that's used in its variable name.

Reply:
:+1: